src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] defaultArray;
+    private int size;
+
+    public ArrayList() {
+        defaultArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private T[] getNewCapacityArray(T[] array) {
+        int oldCapacity = array.length;
+        int newCapacity = oldCapacity + (oldCapacity >> 1);
+        T[] newCapacityArray = (T[]) new Object[newCapacity];
+        System.arraycopy(array, 0, newCapacityArray, 0, oldCapacity);
+        return newCapacityArray;
+    }
+
+    private T[] addValueByIndex(T[] array, int index) {
+        System.arraycopy(array, index, array, index + 1, size - index);
+        return array;
+    }
+
+    private T[] removeValueByIndex(T[] array, int index) {
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        return array;
+    }
```

Comment:
```suggestion
```
You do not need these methods. They have only one line and called once. Also, they do something completely other than expected looking at their names.